lib_eio/path.ml

```diff
@@ -199,6 +199,16 @@ let rename t1 t2 =
     let bt = Printexc.get_raw_backtrace () in
     Exn.reraise_with_context ex bt "renaming %a to %a" pp t1 pp t2
 
+let symlink ?(exists_ok=false) ~target name =
+  let (Resource.T (dir, ops), old_path) = target in
```

Comment:
We should be consistent about the name here (`old_path` vs `target`). Though it's hard to find a good name. The `target` of an operation is usually the thing being changed, but not in the context of symlinks. `~to` would work, except that it's a reserved word.